compiler/src/main/scala/org/scalajs/nscplugin/PrepJSInterop.scala

```diff
@@ -377,6 +377,30 @@ abstract class PrepJSInterop[G <: Global with Singleton](val global: G)
                 |program is unlikely to function properly.""".stripMargin)
           super.transform(tree)
 
+        case tree if tree.symbol == ExecutionContext_global ||
+            tree.symbol == ExecutionContextImplicits_global =>
+          if (scalaJSOpts.warnGlobalExecutionContext) {
+            global.runReporting.warning(tree.pos,
+                """The global execution context in Scala.js is based on JS Promises (microtasks).
+                  |Using it may prevent macrotasks (I/O, timers, UI rendering) from running reliably.
+                  |
+                  |Unfortunately, there is no way with ES6 only to implement a proper
```

Comment:
```suggestion
                  |Unfortunately, there is no way with ES6 only to implement a performant
```

I mean, you could just implement it with `setTimeout(f, 0)`. In fact, maybe this EC implementation should be provided in `scala.scalajs.concurrent.JSExecutionContext`? There's nothing _wrong_ with it, it's just really slow. Happy to PR.

Reply:
It already exists. 🙂 I forget the object name but the name is "timeouts" and it lives next to "promises".

Reply:
So it does! :) In which case, should it be added to the alternatives below?

Reply:
Hmm, I can't find it in the scaladoc. Checking the source now.
https://www.scala-js.org/api/scalajs-library/latest/scala/scalajs/concurrent/index.html

Reply:
Ah, it's there, but just not exposed in an `import`-able form.

Reply:
Good point. I have added another bullet. PTAL